library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -907,14 +907,15 @@ private SqlSelection getWhereClause(final Uri uri, final String where, final Str
             selection.appendClause(Downloads.Impl._ID + " = ?", getDownloadIdFromUri(uri));
         }
         if (uriMatch == BATCHES_ID) {
-            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", ContentUris.parseId(uri));
+            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", String.valueOf(ContentUris.parseId(uri)));
```

Comment:
better just use `contentUri.getLastPathSegment()` to get a string.

You might get `null` instead of `"null"`